### ISSUE TITLE
Enhancement: [Prompt Library] Let users create their own categories

### ISSUE THREAD

**Issue by meroo36** [danny-avila/LibreChat#3353]
### What features would you like to see added?

**Current Behavior:**
Currently, users can only use predefined categories to organize their prompts in the Prompt Library. This limitation restricts users from customizing their prompt organization based on their specific needs and preferences.

**Proposed Behavior:**
Introduce a feature that enables users to create, edit, and delete custom categories for their prompts. This would allow for greater flexibility and personalization in organizing the Prompt Library.

**Use Case:**
 
- Users with specific projects or themes can create relevant categories to keep their prompts organized.
- Users can categorize prompts based on personal or team workflows, enhancing productivity and ease of use.
- Custom categories can help in quickly locating prompts, especially when dealing with a large number of entries.

If this is something the community needs, I would like to contribute to it.

### More details

This can mean couple of changes in the filter part of the prompt library since there could be unlimited numbers of different categories.

### Which components are impacted by your request?

UI, Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mschmidwalnuts** [comment]
Yes highly needed, also the ability to share entire categories ("prompt bundles?") with certain users

**Comment by cbutler-captech** [comment]
Definately need this

**Comment by harrisonhxy** [comment]
Finally came across the issue related to cue words and cue libraries. have a question for you, after I set the cue word, how can the cue word be sent automatically when I select this endpoint, or even better, not show up in the chat box. This way I can start the conversation straight away. 

Currently I have it set up so that I have turned on the button to automatically send prompts, but every time I want to invoke it, I need to `/` first send the prompts that are already set up and wait for the model to reply (since my prompts are so long, the process of sending them feels like an unnecessary conversation, it's too much of a hassle), before I can proceed to the rest of the conversation!

**Comment by danny-avila** [comment]
Triage note on a 2-year-old request with 23 upvotes, one of the highest-demand items left in the aging backlog. Still valid, and cheaper than the thread implies: `category` on the promptGroup schema is already a free-form indexed `String` (`packages/data-schemas/src/schema/promptGroup.ts`), not an enum. Arbitrary category values are therefore storable today. The restriction is entirely in the UI, which offers a fixed list and filters against it.

So this is mostly a UI change plus an endpoint to list a user's distinct categories, which is close to what @meroo36 offered to build in the original post.

@mschmidwalnuts, sharing a whole category is a separate feature that depends on group sharing, and is better tracked alongside the group work than here.

Classification: enhancement, severity low, priority high given the demand and that the reporter volunteered to implement it.
